hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/SnapshotDeletingService.java

```diff
@@ -234,13 +238,26 @@ public BackgroundTaskResult call() throws Exception {
               if (!(toReclaim.getKeyInfosCount() ==
                   repeatedOmKeyInfo.getOmKeyInfoList().size())) {
                 toReclaimList.add(toReclaim.build());
```

Comment:
Can the keys here in `toReclaimList` be reclaimed in-place as well, similar to the logic below?

Reply:
It can be. But one problem is if the OM leader changes in between, it won't have the updated snapshot's `deletedTable` if we do this in-place. As a result, if a follower OM becomes leader it doesn't have the updated 'deletedTable` it would repeat the same splitting of `RepeatedOmKeyInfo` which might be redundant. Ideally the snapshot's `deletedTable` should be cleared in the next iterations.